zebra/kernel_netlink.c

```diff
@@ -386,6 +386,27 @@ static int netlink_socket(struct nlsock *nl, unsigned long groups,
 	return ret;
 }
 
+/* Make socket for Linux generic netlink interface. */
+static int ge_netlink_socket(struct nlsock *nl, ns_id_t ns_id)
```

Comment:
why are we not abstracting and using netlink_socket and why are we not using bind?  I'm not a fan of cut-n-paste coding.  If we need this then let's properly abstract it

Reply:
@donaldsharp Many thanks for your review!

I made the requested change.

1. I abstracted `netlink_socket()`. Now `netlink_socket()` takes an additional argument `nl_family` that specifies the Netlink family of the socket to create. This allows the developer to create any kind of Netlink socket (not only `NETLINK_ROUTE` sockets).
2. I added a new function `rt_netlink_socket()` that can be used to create a `NETLINK_ROUTE` socket. It calls `netlink_socket()` with `nl_family=NETLINK_ROUTE`.
3. I added a new function `ge_netlink_socket()` that can be used to create a `NETLINK_GENERIC` socket. It calls `netlink_socket()` with `nl_family=NETLINK_GENERIC`.

Here is the diff: https://github.com/FRRouting/frr/compare/43430662781371e161d3084d452b5484900aef23..178b584374ae5691a8ce6252faf26550080760b8